composer/trainer/trainer.py

```diff
@@ -778,7 +778,6 @@ def _train_loop(self) -> None:
 
                         # total_loss can be None if gradient scaling failed
                         dist.all_reduce(total_loss, reduce_operation="SUM")
-                        dist.barrier()
```

Comment:
I'm pretty sure this is at least the third time this line has been removed 😂 

(I'm all for removing it)